curriculum/challenges/english/08-coding-interview-prep/rosetta-code/self-describing-numbers.english.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Self Describing Numbers
+id: 5eaf48389ee512d4d103684b
+challengeType: 5
+---
+
+## Description
+<section id='description'>
+There are several so-called "self describing" or <a href="https://en.wikipedia.org/wiki/Self-descriptive_number">"self-descriptive"</a> integers. <br />
+
+An integer is said to be "self-describing" if it has the property that, when digit positions are labeled 0 to N-1, the digit in each postion is eqal to the number of times that that digit appears in the number. <br />
+
+For example, <b>2020</b> is a four-digit self describing number: <br />
+<ul>
+    <li> position 0 has value 2 and there are two 0s in the number; </li>
+    <li> position 1 has value 0 and there are no 1s in the number; </li>
+    <li> position 2 has value 2 and there are two 2s; </li>
+    <li> position 3 has value 0 and there are zero 3s; </li>
+</ul>
+
+Self-describing numbers < 100.000.000  are:     1210,   2020,   21200,   3211000,   42101000.
```

Comment:
We can remove the line breaks and clean up the spacing. Also 100,000,000 can use commas instead of periods:

```
For example, <b>2020</b> is a four-digit self describing number:
<ul>
    <li> position 0 has value 2 and there are two 0s in the number; </li>
    <li> position 1 has value 0 and there are no 1s in the number; </li>
    <li> position 2 has value 2 and there are two 2s; </li>
    <li> position 3 has value 0 and there are zero 3s; </li>
</ul>

Self-describing numbers < 100,000,000 are: 1210, 2020, 21200, 3211000, 42101000.
```